.github/workflows/provider-map-jobs.yml

```diff
@@ -0,0 +1,111 @@
+name: Provider Map Jobs
+on:
+  # TODO: Remove this once behavior has been finalized
+  push:
+    branches:
+      - vlad/provider-map-gha
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * 0"
+
+jobs:
+  fetch_warehouse:
+    name: Fetch Warehouse Updates
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Authenticate with Google
+        uses: google-github-actions/auth@v1
+        with:
+            credentials_json: '${{ secrets.GCP_SA_KEY }}'
+
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v1
+
+      - name: Download providers.csv from warehouse
+        run: |
+          bq query \
+              --quiet \
+              --headless \
+              --format=csv \
+              --use_legacy_sql=false \
+              --max_rows=500 \
+              "SELECT * FROM \`mart_transit_database.dim_mobility_mart_providers\`" \
+          > src/metadata/providers/providers.csv
+
+      - name: Fix our CSV file
+        run: |
+          # Workaround because of...
+          #   https://github.com/google-github-actions/setup-gcloud/issues/666
+          sed -i -n -e '/agency_name/,$p' src/metadata/providers/providers.csv
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install pandas
+        run: |
+          pip install pandas
+
+      - name: Update Counties GeoJSON file
```

Comment:
I feel like this script would be easier to follow/debug/etc. if it was a standalone Python file, say in the `.github/workflows` directory next to this workflow.

Same goes for the requirements above, can we please have a `requirements.txt` that lists out `pandas` and pins a specific version -- and the workflow installs from there?